R/cloud.R

```diff
@@ -527,18 +527,34 @@ cloud_email <- function(type = c("broken", "failed"), job_id = cloud_job(), pkg
 
 #' Return the current cloud job
 #'
-#' This is automatically set by [cloud_check()] and only lasts for the current R session.
+#' The `job_id` is automatically set by [cloud_check()] and is remembered for
+#' the duration of the current R session. If there is no active `job_id`, but
+#' there are local cloud check results, `job_id` is inferred from the most
+#' recently modified cloud check results.
+#'
 #' @param job_id If not `NULL`, sets the active `job_id` to the input.
+#' @inheritParams cloud_report
 #' @export
-cloud_job <- function(job_id = NULL) {
-  if (!is.null(job_id)) {
-    cloud_data$job_id <- job_id
+cloud_job <- function(job_id = NULL, pkg = ".") {
+  cloud_data$job_id <- job_id %||% cloud_data$job_id
+  if (!is.null(cloud_data$job_id)) {
+    return(invisible(cloud_data$job_id))
   }
 
-  if (is.null(cloud_data$job_id)) {
-    stop("No current job, please specify the `job_id` explicitly, or run a job with `cloud_check()`", call. = FALSE)
+  pkg <- pkg_check(pkg)
+  cloud <- dir_find(pkg, "cloud")
+  if (dir.exists(cloud)) {
+    cloud_dirs <- list.dirs(cloud, recursive = FALSE)
+  } else {
+    cloud_dirs <- character()
+  }
```

Comment:
We have to check for the existence of `revdep/cloud.noindex` anyway, so maybe I don't need to move its creation from `cloud_report()` to `cloud_check()` (?).

Reply:
But the `job_id` subfolder won't be there until you actually retrieve some data, so if you fire off a job, then exit from R you don't really have a way to retrieve the job id. If we create the `job_id` subdirectory in `cloud_check()` then with this PR you could just start a new R session and run `cloud_report()` and it would find it.

Reply:
Good point. OK I'll do it.